include/nbl/asset/filters/CSwizzleableAndDitherableFilterBase.h

```diff
@@ -62,29 +62,33 @@ class CSwizzleableAndDitherableFilterBase
 
 			The function supports compile-time decode.
 		*/
-		template<E_FORMAT inFormat, typename Tdec, typename Tenc>
-		static void onDecode(state_type* state, const void* srcPix[4], Tdec* decodeBuffer, Tenc* encodeBuffer, uint32_t blockX, uint32_t blockY)
+
+		template<E_FORMAT inFormat, typename Tdec>
+		static void onDecode(state_type* state, const void* srcPix[4], Tdec* decodeBuffer, uint32_t blockX, uint32_t blockY)
 		{
 			static_assert(sizeof(Tdec)==8u, "Encode/Decode types must be double, int64_t or uint64_t!");
-			static_assert(sizeof(Tenc)==8u, "Encode/Decode types must be double, int64_t or uint64_t!");
 			asset::decodePixels<inFormat>(srcPix, decodeBuffer, blockX, blockY);
```

Comment:
decode to `tmp/swizzled` **you have no guarantee that `decodeBuffer` is 4 elements long